pkg/hooks/hooks.go

```diff
@@ -33,16 +33,22 @@ const HookSocketsSharedDirectory = "/var/run/kubevirt-hooks"
 type HookSidecarList []HookSidecar
 
 type ConfigMap struct {
-	Name string
-	Key  string
+	Name string `json:"name"`
+	Key  string `json:"key"`
+	Hook string `json:"hook"`
+}
+
+var HookMap = map[string]string{
+	"ondefinedomain":  "/usr/bin/onDefineDomain",
+	"precloudinitiso": "/usr/bin/preCloudInitIso",
```

Comment:
I have mixed feelings about this. The `hooks` module does not require a such binaries. This requirement comes from `sidecar-shim`  container image that you'll be using.

At the same time, in the `template.go` you need the place/path where you are going to put the script.

I think the most straightforward way is to put the binary path in the `Hook` field of ConfigMap (or perhaps `HookPath` instead?). This way you don't need to add `sidecar-shim` constrains into the `hooks` module.

Reply:
This makes sense! I'll modify things accordingly.

My only concern here (thinking out loud) would be for the user to remember to specify the path. But that shouldn't be a concern since, I assume, most users would be sysadmin folks. And we can make this clear in the docs.

Reply:
Yes, some knowledge wrt to execution is needed but that was true for anyone using `sidecar-shim` image before too (e.g:  `entrypoint` should be kept intact for the sidecar-shim` binary, plus the binary paths).

From knowing that the script is from `OnDefineDomain` hook to `/usb/bin/onDefineDomain` seems not a big stretch to me